Entity/Components/Stats/EntityStats.gd

```diff
@@ -0,0 +1,61 @@
+extends Resource
+class_name EntityStats
+## Used as a container for modified runtime stats of an Entity.
+##
+## Use StatComponent to get a reference to the current stats on the Entity.
+## We can increase these stats with buffs, items, or equipment. We shouldn't modify the stats
+## directly, but instead create a modified version of the stats whenever we need them. This is so
+## we don't have to worry about reverting the state of the stats whenever the buff/item/equipment
+## is removed. See get_modified_stats on BuffBase for an example.
+
+## The previous statement will probably change, since we want to modify stats at runtime and we revert the changes when effects are removed.
+
+var modification_count: int = 0
+var offense_modifier_dict: StatDictBase = null
+var defense_modifier_dict: StatDictBase = null
```

Comment:
So `offense_modifier_dict` is what we modify for buffs/debuffs that increase/decrease offensive power, while `defense_modifier_dict` is what we modify for buffs/debuffs that increase/decrease defensive power?

How does a buff like `draw 2 more cards` fit into one of these two dictionaries?

Reply:
It will be stored in the offense dictionary.
The reason is that the value modified by stats (whatever the effect and the stats) is calculated like that:
`base_value + caster_attack_stat - target_defense_stat`
So let's say we buff ourselves with 2 more draw. We add 2 to our draw stat in our `offense_modifier_dict`. We then cast draw 1 card on ourselves. The caster is us, and the target is us
We add the `caster_attack_stat` (which is 2), and we deduce the `target_defense_stat` (which is 0 because we didn't touch our `defense_modifier_dict`).
So the total is:
`1 + 2 - 0` which is indeed 3 cards as expected

Reply:
Gotcha. Would it make more sense to update the naming to `caster_modifier_dict` and `target_modifier_dict`?

Reply:
Mmh I don't think it would; it might even be confusing. If you cast a strength buff on yourself, do you modify the caster or the target dict ? It would make sense to modify the target dict since you are targeting a buff onto someone, but turns out you need to modify the caster dict.
Same thing if an enemy applies weakness to you, they are targeting the player, but the effect goes on the caster dict; it's more confusing that way i think